src/plugins/section-category/section-category-utils.ts

```diff
@@ -156,95 +138,57 @@ function createButton(
 
 export function buildPluginState(
   state: EditorState,
-  props: SectionCategoryProps
+  props: EditorProps
 ): PluginState {
   const decorations: Decoration[] = []
   const can = props.getCapabilities()
+  const categories = props.sectionCategories
+  const usedCategoryIDs = getUsedSectionCategoryIDs(state)
 
   state.doc.descendants((node, pos) => {
+    if (
+      node.type === schema.nodes.abstracts ||
+      node.type === schema.nodes.box_element
+    ) {
+      return false
+    }
     if (isSectionNode(node)) {
-      const attrs = node.attrs as SectionNode['attrs']
-      if (
-        isEditableSectionCategoryID(attrs.category as string) &&
-        !isUnique(attrs.category as string) &&
-        !isSubSection(attrs.category as string)
-      ) {
-        decorations.push(
-          Decoration.widget(pos + 1, (view) =>
-            createButton(
-              view,
-              props,
-              node,
-              pos,
-              attrs.category as string,
-              can?.editArticle
-            )
-          )
-        )
+      const categoryID = node.attrs.category
+      if (isSubsection(categoryID)) {
```

Comment:
We can get rid of it, instead we should return "false" at the end of the if statement (to stop descending).

Reply:
Done.